packages/next/lib/helpers/should-use-yarn.ts

```diff
@@ -1,13 +1,20 @@
 import { execSync } from 'child_process'
+import fs from 'fs'
 
 export function shouldUseYarn(): boolean {
   try {
     const userAgent = process.env.npm_config_user_agent
     if (userAgent) {
       return Boolean(userAgent && userAgent.startsWith('yarn'))
+    } else {
+      if (fs.existsSync('yarn.lock')) {
```

Comment:
This should use the application dir to check, otherwise it'll be relative to the CWD.